src/Utils.cc

```diff
@@ -60,15 +60,36 @@ bool loadPose(sdf::ElementPtr _sdf, ignition::math::Pose3d &_pose,
   std::pair<std::string, bool> framePair =
       sdf->Get<std::string>("relative_to", "");
 
+  // Read the rotation type attribute.
+  std::pair<std::string, bool> rotationTypePair =
+      sdf->Get<std::string>("rotation_type", "rpy_radians");
+
   // Read the pose value.
   std::pair<ignition::math::Pose3d, bool> posePair =
     sdf->Get<ignition::math::Pose3d>("", ignition::math::Pose3d::Zero);
 
   // Set output, but only if the return value is true.
   if (posePair.second)
   {
-    _pose = posePair.first;
     _frame = framePair.first;
+
+    // Modify the rotation component based on the rotation type expected.
+    // Only rpy_degrees require a conversion from degrees to radians.
+    if (rotationTypePair.second && rotationTypePair.first == "rpy_degrees")
+    {
```

Comment:
nit It's not clear if you preserve the true values.

Are the RPY values here computed from the rotation? If so, I think that's defective - you should transform the user input, not the post-computed values.

Reply:
I have reverted the changes, https://github.com/ignitionrobotics/sdformat/pull/589/commits/aac7829b64c45ab18f667c715e57c4e0f34fa5cf#diff-7d09d6004075a16fceed8b6665c62249a2f928e86838438073c983b1c2ad6ad7L63-L92

All the parsing happens in value adding process of `Param` now.